tests/component/encapsulation.cpp

```diff
@@ -194,20 +180,35 @@ TEST(Encapsulation, hierarchyCircular)
     libcellml::ComponentPtr child2 = std::make_shared<libcellml::Component>();
     child2->setName("child2");
 
+    // Standard addition of one component onto another.
     parent->addComponent(child1);
+    EXPECT_EQ(size_t(0), child1->componentCount());
+    EXPECT_EQ(size_t(1), parent->componentCount());
+
+    // Can't make this circular hierarchy 'parent' will not be added as a child
+    // of 'child1'. Everything will remain as it is.
     child1->addComponent(parent);
+    EXPECT_EQ(size_t(0), child1->componentCount());
+    EXPECT_EQ(size_t(1), parent->componentCount());
 
+    // Add the 'parent' component onto the model to make a
+    // waterfall hierarchy of two steps.
     model->addComponent(parent);
 
     libcellml::Printer printer;
     std::string a_parent = printer.printModel(model);
     EXPECT_EQ(e_parent_1, a_parent);
 
+    // Add 'child2' to 'child1' to make a waterfall hierarchy of
+    // three steps.
     child1->addComponent(child2);
     a_parent = printer.printModel(model);
     EXPECT_EQ(e_parent_2, a_parent);
 
+    // Try to make a circular hierarchy but we will not succeed as this is not
+    // allowed.  The model will stay as it is.
     child2->addComponent(parent);
+    EXPECT_FALSE(parent->hasAncestor(child2));
     a_parent = printer.printModel(model);
```

Comment:
Are we happy what happens when parsing a model with an invalid encapsulation structure cf creating it through the API?  This change will not allow it to be created, which is different to how we're dealing with all the other kinds of errors

Reply:
No it won't but it can be logged.

Reply:
OK, good :)